components/ai_chat/core/browser/test_utils.cc

```diff
@@ -23,82 +22,90 @@ namespace ai_chat {
-        break;
-      }
-      default:
-        message = base::StrCat({message, "\n - unknown event"});
+  // we don't need to compare the times to the micro second
+  cloned_turn->created_time = TruncateToSeconds(cloned_turn->created_time);
+
+  // Do the same for edits
+  if (cloned_turn->edits) {
+    std::vector<mojom::ConversationTurnPtr> transformed_edits;
+    for (const auto& edit : cloned_turn->edits.value()) {
+      transformed_edits.push_back(CloneForCompare(edit, compare_uuid));
     }
+    cloned_turn->edits = std::move(transformed_edits);
   }
-  return message;
+  return cloned_turn;
 }
 
 }  // namespace
 
 void ExpectConversationEquals(base::Location location,
-                              const mojom::ConversationPtr& a,
-                              const mojom::ConversationPtr& b) {
-  SCOPED_TRACE(testing::Message() << location.ToString());
-  if (!a || !b) {
-    EXPECT_EQ(a, b);  // Both should be null or neither
-    return;
+                              const mojom::ConversationPtr& a_raw,
+                              const mojom::ConversationPtr& b_raw,
+                              bool compare_non_persisted_fields) {
+  auto a = a_raw->Clone();
+  auto b = b_raw->Clone();
+  if (!compare_non_persisted_fields) {
+    // has_content is not persisted
+    a->has_content = false;
+    b->has_content = false;
+    // Date is not persisted
+    a->updated_time = base::Time::Now();
+    b->updated_time = base::Time::Now();
+    // content_id is not persisted
+    for (auto& content : a->associated_content) {
+      content->content_id = 0;
+    }
+    for (auto& content : b->associated_content) {
+      content->content_id = 0;
+    }
   }
-  EXPECT_EQ(a->uuid, b->uuid);
-  EXPECT_EQ(a->title, b->title);
-  EXPECT_EQ(a->has_content, b->has_content);
-
-  // associated content
-  ExpectAssociatedContentEquals(FROM_HERE, a->associated_content,
-                                b->associated_content);
+  SCOPED_TRACE(testing::Message() << location.ToString());
+  EXPECT_MOJOM_EQ(*a, *b);
 }
 
 void ExpectAssociatedContentEquals(
     base::Location location,
     const std::vector<mojom::AssociatedContentPtr>& a,
-    const std::vector<mojom::AssociatedContentPtr>& b) {
+    const std::vector<mojom::AssociatedContentPtr>& b,
+    bool compare_non_persisted_fields) {
   SCOPED_TRACE(testing::Message() << location.ToString());
   ASSERT_EQ(a.size(), b.size());
 
   for (size_t i = 0; i < a.size(); ++i) {
     SCOPED_TRACE(testing::Message()
                  << "Comparing associated content at index " << i);
-    const auto& a_content = a[i];
-    const auto& b_content = b[i];
+    const auto& a_content_raw = a[i];
+    const auto& b_content_raw = b[i];
 
-    if (!a_content || !b_content) {
-      EXPECT_EQ(a_content, b_content);  // Both should be null or neither
+    if (!a_content_raw || !b_content_raw) {
+      EXPECT_EQ(a_content_raw,
+                b_content_raw);  // Both should be null or neither
```

Comment:
This seems more like a ASSERT_EQ

Reply:
I'm actually thinking all these should be `EXPECT_X` because the name of the function is `ExpectAssociatedContentEquals`. Here we have early return, so we have determined the 2 are not equal. The rest of the test can continue and tell the developer what else has failed.

Reply:
In a follow up I think we should change all these to expose custom matchers and then callers can use `EXPECT_THAT` or `ASSERT_THAT` with the custom matchers, or they can use `ASSERT_MOJOM_EQ` or `EXPECT_MOJOM_EQ` directly for other types that don't need any custom comparison operations (e.g. the uuid reset).

Reply:
Makes sense since it is a short circuit comparison.